onnxruntime/core/session/standalone_op_invoker.cc

```diff
@@ -365,29 +389,37 @@ onnxruntime::Status CreateOp(const OrtKernelInfo* info,
     output_args.push_back(arg_ptrs.back().get());
   }
 
-  NodePtr node_ptr = std::make_unique<onnxruntime::Node>(std::string("standalone_") + op_name, op_name, "", input_args, output_args, nullptr, domain);
+  NodePtr node_ptr = std::make_unique<onnxruntime::Node>(std::string("standalone_") + op_name, op_name, "",
+                                                         input_args, output_args, nullptr, domain);
+
   for (int i = 0; i < attr_count; ++i) {
     auto attr_proto = reinterpret_cast<const ONNX_NAMESPACE::AttributeProto*>(attr_values[i]);
     node_ptr->AddAttributeProto(*attr_proto);
   }
 
-  auto kernel_def_builder = KernelDefBuilder::Create();
-  kernel_def_builder->SetName(op_name);
-  kernel_def_builder->SetDomain(domain);
-  kernel_def_builder->SinceVersion(version);
-  auto kernel_def = kernel_def_builder->Build();
+  node_ptr->SetSinceVersion(version);
+
+  auto status = kernel_registry->TryFindKernel(*node_ptr, ep->Type(), type_constraint_map, &kernel_create_info);
+  ORT_RETURN_IF_ERROR(status);
+
+  auto& kernel_def = kernel_create_info->kernel_def;
+  ORT_RETURN_IF_NOT(kernel_def, "Kernel definition was not found for node Domain:'",
+                    node_ptr->Domain(), "' op_type:", node_ptr->OpType());
 
   static std::unordered_map<int, OrtValue> kEmptyValueMap;
   static OrtValueNameIdxMap kEmptyNameMap;
 
-  OpKernelInfo tmp_kernel_info(*node_ptr.get(), *kernel_def, *ep, kEmptyValueMap, kEmptyNameMap, kernel_info->GetDataTransferManager());
+  OpKernelInfo tmp_kernel_info(*node_ptr.get(), *kernel_def, *ep, kEmptyValueMap, kEmptyNameMap,
+                               kernel_info->GetDataTransferManager());
   std::unique_ptr<onnxruntime::OpKernel> op_kernel;
 
   static FuncManager kFuncMgr;
```

Comment:
not related to this PR's changes.
I recall that the `k` prefix is for consts. `kEmptyValueMap` and `kEmptyNameMap` could probably be made const.
not sure if `kFuncMgr` could be made const, as KernelCreateFn takes a non-const `FuncManager&` parameter. if it is indeed a static instance that gets mutated, should there be some synchronization? noticed that there are locks elsewhere (e.g., `NodeRepo::mutex_`).